src/bika/lims/browser/worksheet/views/add_analyses.py

```diff
@@ -32,11 +32,27 @@
 from plone.memoize import view
 from plone.protect import CheckAuthenticator
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from senaite.core.catalog import SETUP_CATALOG
 from senaite.core.catalog import ANALYSIS_CATALOG
 from senaite.core.permissions.worksheet import can_edit_worksheet
 from senaite.core.permissions.worksheet import can_manage_worksheets
 
 
+def getServiceUidsByMethod(method):
+    if not api.is_uid(method):
+        method = api.get_uid(method)
+
+    query = {
+        "portal_type": "AnalysisService",
+        "is_active": True,
+        "method_available_uid": method,
+    }
+    setup_catalog = api.get_tool(SETUP_CATALOG)
+    uids = map(lambda s: s.UID, setup_catalog(query))
```

Comment:
better use this:
`uids = map(api.get_uid, setup_catalog(query)`

Reply:
done